### ISSUE TITLE
[BUG]: Function expecting 2 parameters when 1 specified

### ISSUE THREAD

**Issue by kshefchek** [modularml/mojo#1364]
### Bug description

I am testing out the List implementation here: https://github.com/TeamPuzel/mojo-list/blob/master/list.mojo

with the following map methods

```
    fn map[A: AnyType](self, body: fn(T) capturing -> A) -> List[A]:
        var buf = List[A]()
        buf.reserve_capacity(self.count)
        for item in self: buf.append(body(item))
        return buf^
    
    fn map[A: AnyType](self, body: fn(T) -> A) -> List[A]:
        var buf = List[A]()
        buf.reserve_capacity(self.count)
        for item in self: buf.append(body(item))
        return buf^
```

However with this code

```
from util import List


fn square(x: Int) -> Int:
  return x * x

fn main() raises:
   let list = List[Int]([1, 2, 3, 4])
   let squared_list = list.map(square)
```

I receive this error

```
test.mojo(1, 1): candidate not viable: callee expects 2 input parameters, but 1 was specified
test.mojo(1, 1): candidate not viable: callee expects 2 input parameters, but 1 was specified
(function) fn map[A: AnyType](self: Self, body: fn(T, /) capturing -> List[A]
(function) fn map[A: AnyType](self: Self, body: fn(T, /) -> List[A]
```

Is this expected or a bug?  I was wondering if it was related to https://github.com/modularml/mojo/issues/1307

### System information

```shell
mac os mojo 0.5.0 (6e50a738)
```

**Comment by JoeLoser** [comment]
> @kshefchek I thinks it's exactly #1307, unfortunately.

I think so too.  #1307 just got fixed today and I marked it to be included in the v0.6.1 release which should come out next week sometime.  I've removed the `mojo-stdlib` label as well as this was a compiler problem (not a library one).

In any case, can you please give your example code a try with the new v0.6.1 release next week and report back?  Thanks.

**Comment by ematejska** [comment]
The 0.6.1 release is out.  Please give it a try and re-open if this is still an issue.
